app/python/PythonController.py

```diff
@@ -62,6 +63,48 @@ def put_challenge(id, challenge_data, new_code, new_test):
     response = challenge_update.to_json(best_score=True)
     return response
 
+  def repair_challenge(id, code_repair):
+    
+    if code_repair is None:
+      return {"Error": "No repair provided"}
+    
+    req_challenge = PythonChallengeDAO.get_challenge(id)
+
+    if req_challenge is None:
+      return {"Error": "Challenge not found"}
+
+    challenge = PythonChallenge(challenge_data= req_challenge)
+    repair_challenge = PythonChallengeRepair(challenge, code_repair)
+
+    path_temporary = "public/temp/"
+    #save temporary codes
+    repair_challenge.temporary_save(path_temporary)
+
+    #validate repair
+    result = repair_challenge.is_valid_repair()
+    
+    if 'Error' in result:
```

Comment:
Now its good, just a little thing, but you don't need to change it. 
Maybe if you redefine some methods name of ChallengeRepair, you could use the perform_validation method.